Bubble sort assignment

ShoiraTa opened the pull request:
# Part 1 bubble_sort.rb
The bubble_sort method is an algorithm that takes an array as an argument and returns the array in the sorted  format

# Part 2 bubble_sort_by.rb
The bubble_sort_by method accepts a block,  returns a sorted array. It uses yield inside the method definition to reference the block outside of the method.

Leon-Mbegera committed:
Starter files.

Leon-Mbegera committed:
define bubble_sort method

ShoiraTa committed:
define logic

ShoiraTa committed:
solution

ShoiraTa committed:
readme: define section titles

ShoiraTa committed:
readme: define instructions for buble_sort_by file

ShoiraTa committed:
add bubble sort logic

ShoiraTa committed:
rename array attr to arr

Leon-Mbegera committed:
add yield bubble sort

Leon-Mbegera committed:
readme: add author (Leon) info

ShoiraTa committed:
readme: add code highlights

ShoiraTa committed:
readme: update stats

ShoiraTa committed:
readme: update readme theme to dark

ShoiraTa committed:
add social page links in README

ShoiraTa committed:
add authors' github stats in README

ShoiraTa committed:
add repository copy instructions in README

ShoiraTa committed:
add bubble sort algorithm description in README

ShoiraTa committed:
fix: rubocop err

BrianSammit commented on bubble_sort_by.rb:
```diff
@@ -0,0 +1,18 @@
+  end
+  arr
+end
+sorted_array = bubble_sort_by(%w[hey hello hi]) do |left, right|
+  left.length - right.length
+end
+puts sorted_array
```
From line 15 to line 18 is debugging code. Please remove these lines. :hammer_and_wrench: :smile:

BrianSammit reviewed (changes requested):
CHANGES REQUIRED :recycle: :red_circle: 
-----------------
Hello, @ShoiraTa @Leon-Mbegera you are doing an amazing job on this project :clap: I really like that you split the code into two different files that make it easy to read it.  They are really clean. :100: :1st_place_medal: 
However, there are a few changes to make to complete this project. :hammer_and_wrench: see them below :point_down: :eyes: 
---------------

- [x] Your folder that contains the linters actions are in the wrong order. It should be `.github/workflows` not `.workflows/github`. Please fix it. 
- [x] Improve your Readme file 
- Check the grammar. You have minor errors. Use Grammarly to check if the grammar is correct. 
- On "Instructions for buble_sort.rb " is messy. Use only one pair of backticks to highlight what you what in the Readme. 
- On "To get a local copy follow the steps below:" improve the steps to clone the repository. Use bullet points to make it clear. Remember to make the steps thinking on someone that doesn't know to program. 
- [OPTIONAL] Put a screenshot of the code. that gonna make your code more professional. 

- [x] Remember no debugging instructions in committed code. You have a debugging code, see my inline comments to know what I'm talking about. :eyes: [Here](https://questions.microverse.org/t/debugging-instructions-code/346) you can read about why. 
- [x] Improve your commit messages. You have some small errors on the commit messages. Some of the commits are not descriptive. Some are not in imperative. Some are with error grammar. Please check the commit messages before you make the commit. [Here](https://github.com/trein/dev-best-practices/wiki/Git-Commit-Best-Practices#formatting-rules) you can read about the formatting rules. [Here](https://docs.github.com/en/github/committing-changes-to-your-project/changing-a-commit-message) you can read about how to changes the commit message. Please keep in mind these formating rules in mind for the next commits. 
--------------------
Make these required changes :hammer_and_wrench: :point_up_2: and send it again for review. 
Reachme on SLACK if you have any questions. 
@ Brian Sammit Cruz Rodriguez
HAPPY CODING!!!

![200 (1)](https://user-images.githubusercontent.com/64173189/111804412-0f6dda80-889e-11eb-9694-de16935d1b73.gif)

ShoiraTa committed:
updated README

ShoiraTa committed:
readme: include setting up instructions

ShoiraTa committed:
remove debugging instructions

ShoiraTa committed:
improve the steps to clone the repository

ShoiraTa committed:
clean debugging code

ShoiraTa commented:
Hello, 

We worked on all the comments and improved our repository, 
checked spelling mistakes in the readme; 
changed commit messages following the instructions;
deleted debugging code 

Thanks

cyonii reviewed (approved):
# Status: Approved 💯🟢

Hello team, you have shown great work implementing the requests by the previous TSE.

You can now merge your PR and move on.
Happy coding.

[OPTIONAL] You may want to move the `bubble_sort_by` method into the `bubble_sort.rb` file because the automated test will look for that function in the suggested file.

> ### You can find me on slack 👉 [CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) if you need further assistance or clarification. I'm always available and will be glad to help.
